plugins/inputs/logfile/logfile_test.go

```diff
@@ -453,7 +453,7 @@ func createWriteRead(t *testing.T, prefix string, logFile *LogFile, done chan bo
 		select {
 		case <-done2:
 			t.Log("Child completed before timeout (as expected)")
-		case <-time.After(time.Second * 10):
+		case <-time.After(time.Second * 20):
```

Comment:
MacOS build was failing, increased the time to timeout

Reply:
@adam-mateen @dchappa can someone explain what increasing the timeout here means for the test? From context clues, we expect the goroutine to finish in < X seconds. How bad is a regression in this timing test? What is the likelihood that we have to come back to this value and update it? I think that testing that something completed in a unit test in X time is an addition of flakiness in testing.

Reply:
>// TestLogsFileAutoRemoval verifies when a new file matching the configured
// FilePath is discovered, the old file will be automatically deleted ONLY after
// being read to the end-of-file. Also verifies the new log file is discovered
// before finishing the old file.

In other words, the intent is to have a certain order of events:
1. create first file
2. write lots of data to first file
3. while "the agent" is reading first file, create a 2nd file. it is *important** that the agent is not done reading the first file.
4. verify the agent reports everything from the first file and then deletes it.

It is unlikely that we need to increase the timeout further. Computers generally get faster not slower. If anything we may get to the point there the computer is reading all the data from the first file before we create the 2nd file and write to it.

I am definitely open to rewrites here if you've got a better idea.

AFAIK this test was passing (not flaky) until my most recent change to it. S

Reply:
I don't think we can make it cleaner without rewriting the actual mechanism for tailing.. so it is what it is.